src/proxy/topic/ReliableTopicProxy.ts

```diff
@@ -150,6 +151,61 @@ export class ReliableTopicProxy<E> extends BaseProxy implements ITopic<E> {
         }
     }
 
+    publishAll(messages: any[]): Promise<void> {
+        const reliableTopicMessages: ReliableTopicMessage[] = [];
+        assertNotNull(messages);
+
+        try {
+            for (const message of messages) {
+                const reliableTopicMessage = new ReliableTopicMessage();
+                reliableTopicMessage.payload = this.serializationService.toData(message);
+                reliableTopicMessage.publishTime = Long.fromNumber(Date.now());
+                reliableTopicMessage.publisherAddress = this.localAddress;
+                reliableTopicMessages.push(reliableTopicMessage);
+            }
+            switch (this.overloadPolicy) {
+                case TopicOverloadPolicy.ERROR:
+                    for (const reliableTopicMessage of reliableTopicMessages) {
```

Comment:
Instead of for loops and add() methods of ringbuffer, you should use addAll methods like Java. You also need to await these promises.

Reply:
The thing that confuses me is should I implement publishAll or publishAllAsync?

Reply:
I see, I think done now.